File: SWIFT/Cap07/c07ex04.swift
// c07ex04.swift

import Foundation

func Fatorial( N: Int) {
    var Fat: Int = 1
    for I in 1...N {
        Fat *= I
    }
    print("Fatorial = \(Fat)")
}

var Limite: Int = 0

repeat {
    print("Qual fatorial: ", terminator: "")
    if let entraLimite = Int(readLine() ?? "") {
        Limite = entraLimite
        break
    }
    else {
        print("\nEntrada invalida. Tente novamente.\n")
    }
} while (true)


Fatorial(N: Limite)

print()
print("Tecle <Enter> para encerrar...", terminator: "")
_ = readLine()

File: SCALA/Cap07/c07ex04.scala
// c07ex04.scala

import scala.io.StdIn._

object Main {
  def fatorial(N: Int): Unit = {
    var fat = 1
    for (i <- 1 to N) {
      fat *= i
    }
    println(s"Fatorial = $fat")
  }

  var limite: Int = 0

  def main(args: Array[String]): Unit = {
    print("Qual fatorial: ")
    limite = readInt()

    fatorial(limite)

    print("\nTecle <Enter> para encerrar... ")
    readLine()
  }
}
